Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs

```diff
@@ -2090,44 +2092,98 @@ public void ConfirmWheelslip(float elapsedClockSeconds)
         /// </summary>
         protected virtual void UpdateSteamEjector(float elapsedClockSeconds)
         {
-            if (Simulator.Settings.SimpleControlPhysics)
-            // Simple braking - control Ejector automatically based upon the brake control position
-            // Stop ejector operation if full vacuum pressure reached
-            {
-                if ((TrainBrakeController.TrainBrakeControllerState == ControllerState.Release || TrainBrakeController.TrainBrakeControllerState == ControllerState.FullQuickRelease || (TrainBrakeController.TrainBrakeControllerState == ControllerState.VacContServ)) && (this.BrakeSystem.BrakeLine1PressurePSI > Vac.ToPress(this.TrainBrakeController.MaxPressurePSI)))
-                {
-                    LargeSteamEjectorIsOn = true;  // If brake is set to a release controller, then turn ejector on
-                }
-                else
+                 // Ejectors are controlled independently for the "straight_vacuum_single_pipe" brake type 
+                 // Ejectors are controlled by brake control valves in Simple Physics Control
+                if (Simulator.Settings.SimpleControlPhysics && CarBrakeSystemType != "straight_vacuum_single_pipe")
+                // Simple braking - control Ejector automatically based upon the brake control position
+                // Stop ejector operation if full vacuum pressure reached
                 {
-                    LargeSteamEjectorIsOn = false; // If brake is not set to a release controller, or full vacuum reached, then turn ejector off
+                    if ((TrainBrakeController.TrainBrakeControllerState == ControllerState.Release || TrainBrakeController.TrainBrakeControllerState == ControllerState.FullQuickRelease || (TrainBrakeController.TrainBrakeControllerState == ControllerState.VacContServ)) && (this.BrakeSystem.BrakeLine1PressurePSI > Vac.ToPress(this.TrainBrakeController.MaxPressurePSI)))
+                    {
+                        LargeSteamEjectorIsOn = true;  // If brake is set to a release controller, then turn ejector on
+                        if (!LargeEjectorSoundOn)
+                        {
+                            SignalEvent(Event.LargeEjectorOn);
+                            LargeEjectorSoundOn = true;
+                        }
```

Comment:
Can you give an example of how you think that this would be called in the relevant code sections?

The only other thing that I am uncertain about is that it is a parameter that is "inherited" from vacuum brake class, so I am not sure whether this will create any problems.

Reply:
See the attached patch.
[ryan-vacuumbrake.diff.txt](https://github.com/openrails/openrails/files/5960518/ryan-vacuumbrake.diff.txt)

Reply:
Thanks for the patch file example.

I can't use the bool function **LargeSteamEjectorIsOn** because it is used is some code instances where the sound trigger is not required, so it can't be mixed.

Instead I have created a get/set function using as follows:

`        private bool _LargeEjectorSoundOn = false;
        public bool LargeEjectorSoundOn
        {
            get => _LargeEjectorSoundOn;
            private set
            {
                if (value != _LargeEjectorSoundOn)
                {
                    SignalEvent(value? Event.LargeEjectorOn : Event.LargeEjectorOff);
                    _LargeEjectorSoundOn = value;
                }
            }
       }   `

I have inserted this get/set parameter ok in the MSTSLocomotive.cs file, but when I attempt to use it in the StraightVacuumSinglePipe.cs file, I get an error which suggests that the "**set accessor is not accessible**". It allows me to select the parameter, but not compile with it.

Do you have any advice to overcome this issue?

Reply:
`private set` might be too restrictive. If you need to access that property in an inheriting class, try `protected`. Last resort is to use `public`.

Reply:
I have tried some different combinations of `protected` and `public`, and still doesn't work.

I have attached the current patch file.

[sound_trigger.diff.txt](https://github.com/openrails/openrails/files/5968452/sound_trigger.diff.txt)

Reply:
If you change the access levels of the setters to public, by dropping "private" from `private set` and just using `set`, both should work.

Reply:
I had already tried changing the access level to public, but that caused a level clash.

I have removed private from the set statement, and that appears to work.

So I will upload some modifications.

Reply:
code modified